### ISSUE TITLE
Failed fallback dependency download causes unhandled exception

### ISSUE THREAD

**Issue by vedranmiletic** [mesonbuild/meson#6130]
Consider the case where you depend on GTest like:

`gtest_dep = dependency('gtest', fallback : ['gtest', 'gtest_dep'], required : false)`

If GTest is available, use it. If not, use the fallback. If WrapDB is down or there is no internet connection so fallback isn't available, dependancy isn't required so the build can still go on.

However, with Meson 0.52.0 I get the following unhandled exception:

```
Run-time dependency GTest found: NO (tried pkgconfig and system)
Looking for a fallback subproject for the dependency gtest
Downloading gtest source from https://github.com/google/googletest/archive/release-1.8.1.zip
Traceback (most recent call last):
  File "/usr/lib64/python3.7/urllib/request.py", line 1317, in do_open
    encode_chunked=req.has_header('Transfer-encoding'))
  File "/usr/lib64/python3.7/http/client.py", line 1252, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.7/http/client.py", line 1298, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.7/http/client.py", line 1247, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.7/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.7/http/client.py", line 966, in send
    self.connect()
  File "/usr/lib64/python3.7/http/client.py", line 1414, in connect
    super().connect()
  File "/usr/lib64/python3.7/http/client.py", line 938, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/lib64/python3.7/socket.py", line 707, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/lib64/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "/usr/lib/python3.7/site-packages/mesonbuild/msetup.py", line 245, in run
    app.generate()
  File "/usr/lib/python3.7/site-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/usr/lib/python3.7/site-packages/mesonbuild/msetup.py", line 192, in _generate
    intr.run()
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 4020, in run
    super().run()
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 412, in run
    self.evaluate_codeblock(self.ast, start=1)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 451, in evaluate_statement
    return self.evaluate_if(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 516, in evaluate_if
    self.evaluate_codeblock(i.block)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 436, in evaluate_codeblock
    raise e
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 430, in evaluate_codeblock
    self.evaluate_statement(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 443, in evaluate_statement
    return self.assignment(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 1053, in assignment
    value = self.evaluate_statement(node.value)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 441, in evaluate_statement
    return self.function_call(cur)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 778, in function_call
    return func(node, posargs, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 285, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 155, in wrapped
    ret = f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreterbase.py", line 174, in wrapped
    return f(*wrapped_args, **wrapped_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3088, in func_dependency
    d = self.dependency_impl(name, display_name, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3148, in dependency_impl
    return self.dependency_fallback(display_name, kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 3199, in dependency_fallback
    self.do_subproject(dirname, 'meson', sp_kwargs)
  File "/usr/lib/python3.7/site-packages/mesonbuild/interpreter.py", line 2489, in do_subproject
    resolved = r.resolve(dirname, method)
  File "/usr/lib/python3.7/site-packages/mesonbuild/wrap/wrap.py", line 151, in resolve
    self.get_file()
  File "/usr/lib/python3.7/site-packages/mesonbuild/wrap/wrap.py", line 217, in get_file
    path = self.get_file_internal('source')
  File "/usr/lib/python3.7/site-packages/mesonbuild/wrap/wrap.py", line 361, in get_file_internal
    self.download(what, cache_path)
  File "/usr/lib/python3.7/site-packages/mesonbuild/wrap/wrap.py", line 343, in download
    dhash, tmpfile = self.get_data(srcurl)
  File "/usr/lib/python3.7/site-packages/mesonbuild/wrap/wrap.py", line 300, in get_data
    resp = urllib.request.urlopen(url, timeout=req_timeout)
  File "/usr/lib64/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib64/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib64/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.7/urllib/request.py", line 1360, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib64/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno -2] Name or service not known>
```

I get the same output without `required : false`.

**Comment by scivision** [comment]
Thanks, yes I agree this should not be spilling Python traceback.

**Comment by scivision** [comment]
@vedranmiletic  this is fixed by #6139. Thanks again
